### ISSUE TITLE
Support [lint.isort] group_by_package config

### ISSUE THREAD

**Issue by an0o0nym** [astral-sh/ruff#13389]
Hi,
isort linting works very nice, but I am missing a single option to finish my ruff setup - which is 'group_by_package'.

Would it be easy to implement?

**Comment by MichaReiser** [comment]
Hi @an0o0nym 

Thanks for the feature request. We've been hesitant to add support for more isort options. It's a non goal for us to support all isort options. But I'll keep this open to track the feature request.

**Comment by an0o0nym** [comment]
@MichaReiser  thanks for quick reply. Is there any way I can use isort alongside ruff until its implemented [or not].?

**Comment by charliermarsh** [comment]
You can always use isort alongside Ruff -- just be sure not to enable Ruff's `I` rules.

What does this setting do exactly? It's not obvious from the docs.

**Comment by an0o0nym** [comment]
OK, thanks.


Ideally it should do something like :


```python
from django.http import Http404

from rest_framework.generics import CreateAPIView
from rest_framework.generics import ListAPIView
```

so grouping the top-level packages into its own sections.

**Comment by dhruvmanila** [comment]
If I'm understanding it correctly, I think this could be achieved by [`sections`](https://docs.astral.sh/ruff/settings/#lint_isort_sections) config option to introduce the new sections and then updating the [`section-order`](https://docs.astral.sh/ruff/settings/#lint_isort_section-order) like so:

```toml
["tool.ruff"]
select = [ "I" ]

["tool.ruff.isort"]
section-order = [ "future", "standard-library", "third-party", "first-party", "local-folder", "django", "rest_framework" ]
force-single-line = true

["tool.ruff.isort".sections]
django = [ "django" ]
rest_framework = [ "rest_framework" ]
```

The [`force-single-line`](https://docs.astral.sh/ruff/settings/#lint_isort_force-single-line) can be used to keep the imports separated on their own line similar to your example.

Playground: https://play.ruff.rs/2eb35ec5-05b4-4432-8ac9-999b8d79882a

**Comment by an0o0nym** [comment]
@dhruvmanila sure this would work, but as you mentioned it requires naming each package manually in the config.

**Comment by tonycoco** [comment]
Ruff's isort config should probably get parity so folks can use the isort profiles like Google's styleguide:

```
[tool.ruff.lint.isort]
force-single-line = true
force-sort-within-sections = true
lexicographical = true
line-length = 1000
single-line-exclusions = [
    "collections.abc",
    "six.moves",
    "typing",
    "typing-extensions",
]
order-by-type = false
group-by-package = true
```

This impedes a lot of team's ability to share a single config across projects and for us to adhere to Google's styleguide.
